db/commitlog/commitlog.cc

```diff
@@ -718,42 +757,30 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
                 return cycle(true, true);
             }
         }
-        return make_ready_future<sseg_ptr>(shared_from_this());
-    }
-    future<sseg_ptr> close() {
-        _closed = true;
-        auto s = co_await sync();
-        co_await flush();
-        co_await terminate();
-        _segment_manager->totals.wasted_size_on_disk += (_size_on_disk - file_position());
-        co_return s;
+        return make_ready_future<>();
```

Comment:
If terminate can be called multiple times, it should wait on the previous call.
I'd consider using a shared future, and maybe turn  `_terminated` into an std::optional<shared_future<>> or something along these lines.

Reply:
It can't be called in the sense that user code relies on it. It can only be called twice because we might have both close/terminated a segment via a finish_and_get_new and because we are shutting down (and do close-all). 
In shutdown we will anyway wait for all IO ops to finish, so it is dealt with. That being said, it is not wrong to impose a wait here, so could add a simple "wait for all flushes to finish"...

Reply:
fixed